zeus/optimizer/power_limit.py

```diff
@@ -468,3 +470,150 @@ def _save_profile(self) -> None:
                 ),
             )
         self.logger.info("JIT profiling results saved to '%s'.", str(self.profile_path))
+
+
+# Only import HuggingFace Classes when type checking, to avoid hard dependency on HuggingFace Transformers
+if TYPE_CHECKING:
+    from transformers import (
+        TrainerCallback,
+        TrainingArguments,
+        TrainerState,
+        TrainerControl,
+        PreTrainedModel,
+    )
+
+
+def import_hf_classes():
+    """Attempts to import and return several classes from the `transformers` library.
+
+    This function tries to import the following classes from the `transformers` library:
+    - `TrainerCallback`
+    - `TrainingArguments`
+    - `TrainerState`
+    - `TrainerControl`
+    - `PreTrainedModel`
+
+    Returns:
+        tuple: A tuple containing the imported classes if the `transformers` library is installed.
+        None: If the `transformers` library is not installed, returns None.
+
+    """
+    try:
+        from transformers import (
+            TrainerCallback,
+            TrainingArguments,
+            TrainerState,
+            TrainerControl,
+            PreTrainedModel,
+        )
+
+        return (
+            TrainerCallback,
+            TrainingArguments,
+            TrainerState,
+            TrainerControl,
+            PreTrainedModel,
+        )
+    except ImportError:
+        return None
+
+
+def make_hf(cls: Type[Callback], name: str | None = None) -> Type[TrainerCallback]:
+    """Wrap a `zeus` callback class to be compatible with Hugging Face's Trainer.
+
+    Args:
+        cls (Type[Callback]): The `zeus` callback class to wrap.
+        name (str, optional): The name of the new TrainerCallback class. If None, use `cls.__name__ + "HFWrapper"`.
+
+    """
+    # dynamically import Hugging Face classes, to avoid hard dependency
+    hf_classes = import_hf_classes()
+    if hf_classes is None:
+        raise ImportError(
+            "Hugging Face is not installed. Please install it to use this feature."
+        )
+
+    TrainerCallback, _, _, _, _ = hf_classes  # noqa: N806
+
+    class Wrapper(TrainerCallback):
+        def __init__(self, *args, **kwargs) -> None:
+            self.plo = cls(*args, **kwargs)
```

Comment:
Constructor signature binding missing

Reply:
Also `self.plo` is not a good name, since we want the code here to be generic over optimizers. Maybe use `self.optimizer`.